web.test.tests/web.test.tests/Tests/DepositePageTests.cs

```diff
@@ -140,97 +143,100 @@ public void Income_And_Interest()
         }
 
 
-        [TestCase(0, fin_year_360)]
-        [TestCase(1, fin_year_360)]
-        [TestCase(16, fin_year_360)]
-        [TestCase(17, fin_year_360)]
-        [TestCase(169, fin_year_360)]
-        [TestCase(170, fin_year_360)]
-        [TestCase(229, fin_year_360)]
-        [TestCase(360, fin_year_360)]
-        [TestCase(365, fin_year_365)]
-        public void End_Date(int term, String fin_year)
+        [TestCase(0, 360)]
+        [TestCase(1, 360)]
+        [TestCase(16, 360)]
+        [TestCase(17, 360)]
+        [TestCase(169, 360)]
+        [TestCase(170, 360)]
+        [TestCase(229, 360)]
+        [TestCase(360, 360)]
+        [TestCase(365, 365)]
+        public void End_Date(int term, int fin_year)
         {
             //Arrange
             int start_day = 15;
             int start_month = 7;
             int start_year = 2019;
 
             //Act
-            driver.FindElement(By.XPath("//div[text()='Settings']")).Click();
-            new WebDriverWait(driver, TimeSpan.FromMilliseconds(10000)).Until(ExpectedConditions.VisibilityOfAllElementsLocatedBy(By.XPath("//div[text()='Logout']")));
-            SettingsPage settingsPage = new SettingsPage(driver);
+            DepositPage depositPage = new DepositPage(driver);
+            depositPage.OpenSettingsPage();
 
-            DateTime start_date = new DateTime(start_year, start_month, start_day);
-            String exp_end_date = start_date.AddDays(term).ToString(settingsPage.SelectedDateFormat, CultureInfo.InvariantCulture);
-            
+            SettingsPage settingsPage = new SettingsPage(driver);
+            String exp_end_date = new DateTime(start_year, start_month, start_day).AddDays(term).ToString(settingsPage.SelectedDateFormat, CultureInfo.InvariantCulture);
             settingsPage.CancelBtn.Click();
 
-            driver.FindElement(By.XPath($"//*[text()='{fin_year}']/input")).Click();
-
-            SelectElement year_select = new SelectElement(driver.FindElement(By.Id("year")));
-            year_select.SelectByText(start_year.ToString());
-
-            SelectElement month_select = new SelectElement(driver.FindElement(By.Id("month")));
-            month_select.SelectByIndex(start_month - 1);
-
-            SelectElement day_select = new SelectElement(driver.FindElement(By.Id("day")));
-            day_select.SelectByText(start_day.ToString());
-
-            IWebElement term_field = driver.FindElement(By.Id("term"));
-            term_field.Clear();
-            term_field.SendKeys(term.ToString());
+            if (fin_year == 360)
+            {
+                depositPage.FinYear360.Click();
+            }
+            else if (fin_year == 365)
+            {
+                depositPage.FinYear365.Click();
+            }
+            else
+            {
+                Assert.AreEqual("", "Inproper fin_year value in input data");
+            }
```

Comment:
Move this to method on DepositePage. It would be more 'readable'

```
public void SelectFinYear(int year)
{
    switch (year)
    {
        case 360:
            FinYear360.Click();
            break;
        case 365:
            FinYear360.Click();
            break;
        default:
            throw new Exception($"Not valid fin year value: {year}");
    }
}
```

Reply:
Done